### PR TITLE
add task-schema doc

### PR DESCRIPTION
**Pull request by yyscamper**
Add doc for the task schema.

Depends on: 
- https://github.com/RackHD/on-tasks/pull/290
- https://github.com/RackHD/on-core/pull/186

NOTE: The doc reflects the new change in the dependent PRs, so please merge this one after all dependent PRs have been merged.

@RackHD/corecommitters @heckj @amymullins @WangWinson @iceiilin @cgx027 @pengz1

### CODE REVIEW

**Comment by anhou** [comment]
:+1:

File: docs/rackhd/tasks.rst [hunk 1]
```diff
@@ -150,6 +151,169 @@ The primary difference between the *Install CoreOS* task and the *Install Ubuntu
 is the profile value, which is the ipxe template that specifies the installer
 images that an installation target should download.
 
+.. _`Task Schema`:
+
+Task Schema
+^^^^^^^^^^^^^^^^^^^^^^
+A Task Schema is a JSON-Schema_ file that outlines the attributes and validation requirement for all options of a task. It provides standardlized and declarative way to annotate task options. It offloads job's validation work and brings benefit to the upfront validation for graph input options.
```
Comment (WangWinson) [inline docs/rackhd/tasks.rst]: -> standardized

File: docs/rackhd/tasks.rst [hunk 2]
```diff
@@ -150,6 +151,169 @@ The primary difference between the *Install CoreOS* task and the *Install Ubuntu
...
+
+.. _JSON-Schema: http://json-schema.org/
+
+**Relation Between Job & Schema & Task**
+
+- A Task Definition conforms to only one Task Schema, but different Task Definition can conform to the same Task Schema; For example, all obm-control tasks share the same obm-control schema.
+- A Task Schema only describes one Job, but a Job may be described by multiple Task Schemas. For example, All of install-centos, install-ubuntu, install-esxi schemas are to describe the same install-os job.
```
Comment (WangWinson) [inline docs/rackhd/tasks.rst]: -> For example, all of

File: docs/rackhd/tasks.rst [hunk 3]
```diff
@@ -150,6 +151,169 @@ The primary difference between the *Install CoreOS* task and the *Install Ubuntu
...
+.. code-block:: JSON
+
+    {
+        "$schema": "rackhd-task-schema.json",
+        "copyright": "Copyright 2016, EMC, Inc.",
+        "title": "Analyze OS Repository",
+        "description": "The schema for analyzing os repository job",
+        "describeJob": "Job.Os.Analyze.Repo",
+        "allOf": [
+            { "$ref": "common-task-options#/definitions/Options" },
```
Comment (WangWinson) [inline docs/rackhd/tasks.rst]: .json

File: docs/rackhd/tasks.rst [hunk 4]
```diff
@@ -150,6 +151,169 @@ The primary difference between the *Install CoreOS* task and the *Install Ubuntu
...
+                    "osType": {
+                        "readoly": true
+                    }
+                },
+                "required": ["osType", "repo", "version"]
+            }
+        ]
+    }
+
+Above schema example shows its task specific options are *"verion"*, *"repo"* and *"osType"*, it describes the job *Job.Os.Analyze.Repo*.
```
Comment (WangWinson) [inline docs/rackhd/tasks.rst]: version

File: docs/rackhd/tasks.rst [hunk 5]
```diff
@@ -150,6 +151,169 @@ The primary difference between the *Install CoreOS* task and the *Install Ubuntu
...
+        injectableName: 'Task.Os.Esx.Analyze.Repo',
+        implementsTask: 'Task.Base.Os.Analyze.Repo',
+        schemaRef: 'analyze-os-repo.json',
+        options: {
+            osName: 'esx',
+        },
+        properties: {}
+    };
+
+You can define the default value in the *options* property. These default value will be used as complement if user doesn't pass any value for that option while trigger the task, so all default value should conform to the schema as well.
```
Comment (WangWinson) [inline docs/rackhd/tasks.rst]: These default *values*

**Comment by WangWinson** [comment]
:+1:

**Comment by yyscamper** [comment]
Thanks @WangWinson, you have sharp eyes, just like Wukong :) 


I will fix all typos.

**Comment by yyscamper** [comment]
@WangWinson : comment addressed.

**Comment by WangWinson** [comment]
Great ! thx!

**Pull request merged**